internal/gatewayapi/translator.go

```diff
@@ -166,6 +166,9 @@ func newTranslateResult(gateways []*GatewayContext,
 func (t *Translator) Translate(resources *Resources) (*TranslateResult, error) {
 	// Get Gateways belonging to our GatewayClass.
 	gateways := t.GetRelevantGateways(resources.Gateways)
+	for _, gtw := range gateways {
```

Comment:
can this be moved into https://github.com/envoyproxy/gateway/blob/33fceb05ca611e30107ff841cac9d7762eea2048/internal/gatewayapi/contexts.go#L29 ?

I know the naming isnt great 🙈 , but it looks like thats the place where the gatewayContext is being initialized

Reply:
I originally put this call in `t.GetRelevantGateways()` and need to update the parameter from `gateways []*gwapiv1.Gateway` to `resources *Resources`. As a result we have to also change the `TranslatorManager` interface. 

It seems that `TranslatorManager` is unused. Can we remove it and put `t.attachEnvoyProxy()` in `t.GetRelevantGateways()` beside `gc.ResetListeners()`? 🤔

Reply:
you could change the signature for `GetRelevantGateways` https://github.com/envoyproxy/gateway/blob/33fceb05ca611e30107ff841cac9d7762eea2048/internal/gatewayapi/translator.go#L63 to accept `resources *Resources` instead, and then move this logic inside `gc.ResetListeners(resource)`